### ISSUE TITLE
Move NuGet package validation to standard infrastructure

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#15442]
Right now we have some one-off tests we use to validate our NuGet packages are ready and actually "work" end-to-end. There's a number of problems with them today:

1. They reinvent some core integration testing primitives we already have, so they're maintenance overhead.
2. They don't run during regular integration test runs, so we don't discover issues until after they've been broken.
3. There are many manual steps.

We should fix all these.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski do we still need this issue?

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi Yes, this is actually still applicable. We're still running our tests manually which we should just automate (which shouldn't be hard to do.)

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski can we assign this to an actual milestone to get this done in?   Having bugs for 9 years that we don't act on isn't helpful.
